### ISSUE TITLE
Make the `disassemble` and the `nearpc` commands have the same output style for the symbol names

### ISSUE THREAD

**Issue by lebr0nli** [pwndbg/pwndbg#1608]
The result of `nearpc` doesn’t respect `set print asm-demangle on/off`, so the symbol name it showed will not same as `disassemble` for some symbols.
To change the result of `nearpc` showed, now we only can do `set print demangle on/off` because it affected the result of `info symbol ...`
But there’s a problem.
Since we cache the symbol name and only reset it when a new objfile is present, the old results we cached will make the output inconsistent.
(I mean something like this will happen:
<img width="1197" alt="poc" src="https://user-images.githubusercontent.com/61896187/223016345-3c021c8b-bc55-435f-bfff-d0caaf5c9ac7.png">

**Comment by disconnect3d** [comment]
Can we just do something like: `with asm_demangle_on(): info symbol ...` in `symbol.get` and make the demangling always work? (we can also make this configurable like pwndbg-demangle-symbols or sth)

**Comment by lebr0nli** [comment]
> Can we just do something like: `with asm_demangle_on(): info symbol ...` in `symbol.get` and make the demangling always work? (we can also make this configurable like pwndbg-demangle-symbols or sth)

Yes, we can. But we still need to find a way to handle different caches for different values of `print demangle`, to avoid the thing happening in the image I provide.
Here's my attempt to sync `nearpc` with the demangling config for now: https://github.com/lebr0nli/pwndbg/commit/c9df6551fa3db779971b1084368b1322927ffa53
But my approach in that commit is dirty, we might need a better way to do it.

**Comment by disconnect3d** [comment]
cc @OBarronCS
